torch/sparse/semi_structured.py

```diff
@@ -0,0 +1,318 @@
+from collections import namedtuple
+import warnings
+
+import torch
+
+
+__all__ = [
+    "to_sparse_semi_structured",
+    "SparseSemiStructuredTensor",
+]
+
+CUSPARSELT_CONFIG = namedtuple("CUSPARELT_CONFIG", "compression_factor n min_size")
```

Comment:
Note that this "looks public" as it doesn't start with `_` but  is not in the `__all__` list. Which is inconsistent.
You should prepend `_` to all APIs that shouldn't be public.